__tests__/api.test.js

```diff
@@ -70,8 +69,85 @@ describe('GET - /api/articles/:article_id', () => {
         );
       });
   });
-})
 
+  test('GET comment count- ', () => {
+    return request(app)
+      .get(`/api/articles/9`)
+      .expect(200)
+      .then((res) => {
+        expect(res.body.article).toEqual(
+          expect.objectContaining({ comment_count: expect.any(String) })
+        );
+        expect(res.body.article.comment_count).toBe('2');
+      });
+  });
+});
+
+describe("6. PATCH /api/articles/:article_id", () => {
+  test("responds with status 200 and specific article is updated", () => {
+    return request(app)
+      .patch("/api/articles/2")
+      .send({ inc_votes: 50 })
+      .expect(200)
+      .then((res) => {
+        expect(res.body.article).toEqual(
+          expect.objectContaining({
+            article_id: 2,
+            title: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            topic: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number)
+          })
+        );
+      });
+  });
+  
+});
+
+
+describe("8. GET /api/articles?topic", () => {
+    test("200: returns an array of all articles with correct comment count (sorted by date descending as default) when passed no queries", async () => {
+      const res = await request(app)
+      .get("/api/articles").expect(200);
+      expect(res.body.articles.length).toBe(12);
+      expect(res.body.articles).toBeSortedBy("created_at", {
+        descending: true,
+      });
+      res.body.articles.forEach((article) => {
+        console.log(Object.keys(article))
+        expect(Object.keys(article)).toEqual([
+          "article_id",
+          "title",
+          "topic",
+          "author",
+          "body",
+          "created_at",
+          "votes",
+          "comment_count"
+        ]);
+      });
+    });
+    test('200: returns array filtered by topic when given a topic query', async () => {
+      const res = await request(app)
+          .get('/api/articles?topic=mitch')
+          .expect(200)
+      expect(res.body.articles.length).toBe(11);
+      expect(res.body.articles).toBeSortedBy('created_at', { descending: true });
+  });
+  
+  test('404: returns empty array when given topic query that does not exist', async () => {
+    const res = await request(app)
+      .get("/api/articles?topic=BIGMANHARRY")
+        .expect(404);
+        expect(res.body.msg).toEqual("Articles not found");
+  
+  });
+  

```

Comment:
you've checked for a topic that doesnt exist but have you thought about one that exists but doesnt have any articles, what would you expect to return then, probably not an error